docs/user/faq.rst

```diff
@@ -584,6 +585,43 @@ In the meantime, the workaround is to percent-encode the forward slash. If you
 don’t control the clients and can't enforce this, you can implement a Falcon
 middleware component to rewrite the path before it is routed.
 
+How do I adapt my code to default context type changes in Falcon 2.0?
+---------------------------------------------------------------------
+
+The default request/response context type has been changed from dictionary to a
+class subclassing dict in Falcon 2.0. Instead of setting dictionary items, one
+shall set attributes on the object now:
+
+.. code:: python
+
+   # Before Falcon 2.0
+   req.context['cache_backend'] = MyUltraFastCache.connect()
+
+   # Falcon 2.0
+   req.context.cache_backend = MyUltraFastCache.connect()
+
+Since the default context type derives from dict, the existing code will work
+unmodified with Falcon 2.0. Nevertheless, it is recommended to change the usage
+as recommended above since the ability to use context as dictionary may be
+removed in the future major releases.
```

Comment:
Let's include a warning here as well about mixing-and-matching mapping vs. attributes.

Reply:
"in the future major releases" ➡️ "in a future release"